File: app/src/main/java/com/leonov_dev/currencyconverter/ui/home/stock/CurrencyListBinding.java
package com.leonov_dev.currencyconverter.ui.home.stock;

import androidx.databinding.BindingAdapter;

import android.widget.ImageView;
import android.widget.ListView;

import com.leonov_dev.currencyconverter.data.CurrencyReplacement;

import java.util.List;

public class CurrencyListBinding {

    @BindingAdapter("app:items")
    public static void setItems(ListView listView, List<CurrencyReplacement> items) {
        CurrencyReplacementAdapter adapter = (CurrencyReplacementAdapter) listView.getAdapter();
        if (adapter != null)
        {
            adapter.replaceData(items);
        }
    }

    @BindingAdapter("app:flagLogo")
    public static void setFlagLogo(ImageView view, int resId){
        view.setImageResource(resId);
    }

}

File: app/src/main/java/com/leonov_dev/currencyconverter/ui/home/stock/CurrencyListBinding.kt
package com.leonov_dev.currencyconverter.ui.home.stock

import android.view.View
import android.widget.EditText
import androidx.databinding.BindingAdapter

import android.widget.ImageView
import android.widget.ListView

import com.leonov_dev.currencyconverter.data.CurrencyReplacement


@BindingAdapter("app:items")
fun setItems(listView: ListView, items: List<CurrencyReplacement>) {
    val adapter = listView.adapter as CurrencyReplacementAdapter?
    adapter?.replaceData(items)
}

@BindingAdapter("app:flagLogo")
fun setFlagLogo(view: ImageView, resId: Int) {
    view.setImageResource(resId)
}

@BindingAdapter("app:onFocusChange")
fun onFocusChange(text: EditText, listener: View.OnFocusChangeListener) {
    text.onFocusChangeListener = listener
}
